src/hooks/useBalancesList.ts

```diff
@@ -91,12 +91,16 @@ export const useBalancesList = ({
   }, [fetchAccountBalances, publicKey, network]);
 
   // Convert balances object to array
-  const balanceItems = Object.entries(pricedBalances).map(
-    ([id, balance]) =>
-      ({
-        id,
-        ...balance,
-      }) as PricedBalance & { id: string },
+  const balanceItems = useMemo(
```

Comment:
This change is not related to the bug but it improves app performance by preventing unnecessary re-renders due to changes in pricedBalances object refs